### DISCUSSION TITLE
Project Tiers & maplibre-gl-native

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre#186] [Q&A]
Right now the [project tiers](https://github.com/maplibre/maplibre/blob/32edd53ef1f32742aa2b241487a20dd8fd596c37/PROJECT_TIERS.md) are on a repository level.

However, since maplibre-gl-native uses a monorepo (as of now) this includes experimental projects such as [node](https://github.com/maplibre/maplibre-gl-native/tree/main/platform/node) and the community maintained [Qt](https://github.com/maplibre/maplibre-gl-native/tree/main/platform/qt) platform as well.

It would thus be helpful if the project tiers of the various projects contained in the [maplibre-gl-native](https://github.com/maplibre/maplibre-gl-native) repository are clarified a bit.

**Comment by louwers** [comment]
These repositories should also be part of the core:

https://github.com/maplibre/maplibre-plugins-android
https://github.com/maplibre/maplibre-gl-native-distribution

Because they are essential for the Android and iOS platforms.

**Reply by louwers** [reply]
We can also more granularly define which functionality is considered core for maplibre-plugins-android.

**Reply by wipfli** [reply]
Please make a pull request to maplibre/maplibre with the suggested change.

**Reply by fynngodau** [reply]
My perspective on the `maplibre-plugins-android` repository is as follows. Specifically I have been working with the annotations plugin.

Built separately from MapLibre itself, it obviously doesn't include any features that couldn't be implemented by a developer themselves. However, a user of the SDK would expect "convenience" functions such as "add marker to map" to be readily available. Such functions are also available in MapLibre (partially with duplicate code – once written to imitate Google Maps' API), however they are all deprecated in favor of the annotations plugin.

I would recommend the following plan:

* remove deprecated annotation functionality from Android's maplibre-gl-native
* make maplibre-plugins-android a Core project (at least a Supported project)
* think about where it is possible to add more convenience and user-friendliness to the annotations plugin, for example through Kotlin extension functions.

I haven't worked with the other plugins and cannot really comment; it really might make sense to differentiate the tiers per plugin, as some are less important than others.
